### PR TITLE
Atmel Saml2x dts fix

### PR DESCRIPTION
**Pull request by jmalbers**
Updates Atmel SAML device tree files saml2x.dtsi and saml21.dtsi

Many devices do not share registers between saml21 and saml22 socs. 

All devices without shared addresses were moved into the saml21.dtsi

Incorrect power manager address was corrected.

Timer counter 0 - 3 were added to saml2x device tree include.

Built and ran blinky example and hello world.

### CODE REVIEW

**Review by stephanosio — CHANGES_REQUESTED** [review]
Please review the [Contribution Guidelines](https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-guidelines) and re-format the commit message.

**Comment by jmalbers** [comment]
Thanks for the help. First timer here :)

**Comment by nandojve** [comment]
Hi @jmalbers ,

Thank you for look on this. I noted that there are peripherals that you are moving which has same register address. The devicetree should keep shared address at only one file and describe the difference in the more specific file. Try follow [samv71q21b.dtsi](https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/arm/atmel/samv71q21b.dtsi) as referece.
You need structure the change using same principle.

**Comment by jmalbers** [comment]
> Hi @jmalbers ,
> 
> Thank you for look on this. I noted that there are peripherals that you are moving which has same register address. The devicetree should keep shared address at only one file and describe the difference in the more specific file. Try follow [samv71q21b.dtsi](https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/arm/atmel/samv71q21b.dtsi) as referece. You need structure the change using same principle.

Missed a few I see! Thanks for pointing this out :)

File: dts/arm/atmel/saml2x.dtsi [hunk 1]
```diff
@@ -45,160 +37,41 @@
...
 		};
 
-		trng: random@42003800 {
-			compatible = "atmel,sam-trng";
-			reg = <0x42003800 0x24>;
-			peripheral-id = <0>;
-			interrupts = <27 0>;
-		};
 	};
+
```
Comment (nandojve) [inline dts/arm/atmel/saml2x.dtsi:74]: This line is not necessary

File: dts/arm/atmel/saml21.dtsi [hunk 2]
```diff
@@ -7,29 +7,205 @@
...
+
 	soc {
-		usb0: usb@41000000 {
-			compatible = "atmel,sam0-usb";
-			status = "disabled";
-			reg = <0x41000000 0x1000>;
-			interrupts = <6 0>;
-			num-bidir-endpoints = <8>;
+
+		nvmctrl: nvmctrl@41004000  {
```
Comment (nandojve) [inline dts/arm/atmel/saml21.dtsi:22]: nvmctrl@41004000 is the same for L21/22. There is a big error at Figure 8-1. SAM L22 Product Mapping datasheet on the AHB-APB Bridge B part. The prefix of AHB-APB Bridge B is 0x41xx-xxxx and there are entries with 0x40xx-xxxx which is complete wrong.

File: dts/arm/atmel/saml21.dtsi [hunk 3]
```diff
@@ -7,29 +7,205 @@
...
+			reg = <0x41004000 0x22>;
+			interrupts = <4 0>;
+			lock-regions = <16>;
+
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			flash0: flash@0 {
+				compatible = "soc-nv-flash";
+				reg = <0 0x40000>;
```
Comment (nandojve) [inline dts/arm/atmel/saml21.dtsi:33]: flash@0 reg property should be defined on a file named saml21x18.dtsi
idem with sram0: memory@20000000, the reg property should be at saml21x18.dtsi file too.

Same principle of https://github.com/zephyrproject-rtos/zephyr/blob/32f4b6e09a85c4886767df93d8a6c4297b79b009/dts/arm/atmel/samd5xx20.dtsi#L7-L22

**Review by nandojve — CHANGES_REQUESTED** [review]

**Comment by nandojve** [comment]
Hi @jmalbers ,

I was wondering if you had time to look comments. I would like to have it for next release but a few work still needed. Please, let us know if you still have interest and time to finish work.

**Comment by jmalbers** [comment]
Sorry, let this sit for a while. I'll have a look at it again this week.

**Comment by nandojve** [comment]
Hi @jmalbers ,

You need rebase and keep your commits on top of main without merge commits.
May you update?

**Comment by nandojve** [comment]
Hi @jmalbers ,

Merge windows is open again and we can continue work here.

**Pull request closed without merging**